### PR TITLE
Add support for creating/deleting a delegation to Notary Repository

### PR DESCRIPTION
**Pull request by cyli**

### CODE REVIEW

File: client/client_test.go [hunk 1]
```diff
@@ -1475,3 +1475,126 @@ func TestRemoteServerUnavailableNoLocalCache(t *testing.T) {
 	assert.Error(t, err)
 	assert.IsType(t, store.ErrServerUnavailable{}, err)
 }
+
+// AddDelegation creates a valid changefile (rejects invalid delegation names),
+// which when applied, adds a new delegation role with the correct keys.
+func TestAddDelegation(t *testing.T) {
```
Comment (diogomonica) [inline client/client_test.go]: Should we try multi-level delegation? `targets/a/b/c`?

Comment (cyli) [inline client/client_test.go]: To verify that AddDelegation doesn't validate the hierarchy, you mean?  Sounds good, will change it.  :)

File: client/helpers_test.go [hunk 2]
```diff
@@ -728,3 +728,37 @@ func TestApplyTargetsDelegationCreate2Deep(t *testing.T) {
 	assert.Equal(t, "targets/level1/level2", role.Name)
 	assert.Equal(t, "level1/level2", role.Paths[0])
 }
+
+func TestApplyTargetsDelegationInvalidParent(t *testing.T) {
```
Comment (diogomonica) [inline client/helpers_test.go]: Why is this invalid?

File: client/client_test.go [hunk 3]
```diff
@@ -1475,3 +1475,126 @@ func TestRemoteServerUnavailableNoLocalCache(t *testing.T) {
...
+	newDelegationRole := targetRole.Signed.Delegations.Roles[0]
+	assert.Len(t, newDelegationRole.KeyIDs, 1)
+	assert.Equal(t, targetPubKey.ID(), newDelegationRole.KeyIDs[0])
+	assert.Equal(t, "targets/a", newDelegationRole.Name)
+}
+
+// RemoveDelegation rejects attempts to remove invalidly-named delegations,
+// but otherwise does not validate the name of the delegation to remove.  This
+// test ensures that the changefile generated by RemoveDelegation is correct.
+func TestRemoveDelegationChangefileValid(t *testing.T) {
```
Comment (diogomonica) [inline client/client_test.go]: Multi-level here too.

**Comment by diogomonica** [comment]
LGTM

File: client/helpers_test.go (hunk 2 above)
Comment (cyli) [inline client/helpers_test.go]: Sorry, non-existent.  Will change.  :)

File: client/client_test.go (hunk 3 above)
Comment (cyli) [inline client/client_test.go]: Changed in c917c0b

**Comment by endophage** [comment]
LGTM

**Pull request merged**